### PR TITLE
Fill in Publish / Request Header Gaps #839

### PR DESCRIPTION
**Pull request by scottf**
addressing https://github.com/nats-io/nats.net/issues/736

### CODE REVIEW

File: src/NATS.Client/IConnection.cs [hunk 1]
```diff
@@ -127,14 +129,7 @@ public interface IConnection : IDisposable
...
         void Publish(string subject, byte[] data, int offset, int count);
 
-        /// <summary>
-        /// Publishes a <see cref="Msg"/> instance, which includes the subject, an optional reply, and an
-        /// optional data field.
-        /// </summary>
-        /// <param name="msg">A <see cref="Msg"/> instance containing the subject, optional reply, and data to publish
-        /// to the NATS server.</param>
-        /// <seealso cref="IConnection.Publish(string, byte[])"/>
-        void Publish(Msg msg);
```
Comment (scottf) [inline src/NATS.Client/IConnection.cs:137]: this is just moved lower down for organizational purposes

File: src/NATS.Client/IConnection.cs [hunk 2]
```diff
@@ -399,41 +463,7 @@ public interface IConnection : IDisposable
...
-        /// subscription for all replies to this <see cref="Connection"/> instance. However, if 
-        /// <see cref="Options.UseOldRequestStyle"/> is set, each request will have its own underlying subscription. 
-        /// The old behavior is not recommended as it may cause unnecessary overhead on connected NATS servers.
-        /// </remarks>
-        /// <param name="message">A NATS <see cref="Msg"/> that contains the request data to publish
-        /// to the connected NATS server.  The reply subject will be overridden.</param>
-        /// <param name="timeout">The number of milliseconds to wait.</param>
-        /// <returns>A <see cref="Msg"/> with the response from the NATS server.</returns>
-        /// <seealso cref="IConnection.Request(Msg)"/>
-        Msg Request(Msg message, int timeout);
```
Comment (scottf) [inline src/NATS.Client/IConnection.cs:436]: this is just moved up for organizational purposes

**Comment by ColinSullivan1** [comment]
Is there a request for this? It is convenient but the API footprint if getting large with all of the overloads. In the past I nudged users toward msg API if they needed to use headers.

**Review by ColinSullivan1 — APPROVED** [review]
Per direct discussions, the API footprint addition falls into an ease of use feature so it a positive imo.  LGMT.

**Pull request merged**